specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-09-30-preview/openShiftManagedClusters.json

```diff
@@ -0,0 +1,650 @@
+        },
+        "provider": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource provider for the operation."
+        }
+      },
+      "description": "Describes the properties of a Compute Operation Value Display."
+    },
+    "Resource": {
+      "description": "The Resource model definition.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "PurchasePlan": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The plan ID."
+        },
+        "product": {
+          "type": "string",
+          "description": "Specifies the product of the image from the marketplace. This is the same value as Offer under the imageReference element."
+        },
+        "promotionCode": {
+          "type": "string",
+          "description": "The promotion code."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "The plan ID."
+        }
+      },
+      "description": "Used for establishing the purchase context of any 3rd Party artifact through MarketPlace."
+    },
+    "OpenShiftRouterProfile": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the router profile."
+        },
+        "publicSubdomain" : {
+          "type": "string",
+          "description": "DNS subdomain for openshift router."
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Auto-allocated FQDN for the OpenShift router."
+        }
+      },
+      "description": "Represents an OpenShift router"
+    },
+    "ContainerServiceVnetSubnetID": {
+      "type": "string",
+      "description": "VNet SubnetID specifies the vnet's subnet identifier."
+    },
+    "OSType": {
+      "type": "string",
+      "default": "Linux",
+      "enum": [
+        "Linux",
+        "Windows"
+      ],
+      "x-ms-enum": {
+        "name": "OSType",
+        "modelAsString": true
+      },
+      "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
```

Comment:
When I am trying to change the description I have this error : 
```
Process() cancelled due to exception : '$.definitions.OSType.description' has incompatible values (---
Specifies the OS type.
, ---
  -
  OsType to be used to specify os type. Choose from Linux and Windows. Default
  to Linux.
).
  Error: '$.definitions.OSType.description' has incompatible values (---
Specifies the OS type.
, ---
  -
  OsType to be used to specify os type. Choose from Linux and Windows. Default
  to Linux.
).
```

Reply:
is this because the `OSType` is also used in the other swagger file ?

Reply:
Changing the content of the string shouldn't alter the build.... are you sure it looks like this:
```
"description": "Specifies the OS type. Options are Linux and Windows. The default is Linux."
```

Having said that - if you *do* have the same enum else where in the spec(s), then you will need to ensure:
- They have the same values.
- They will always have the same values (i.e. one wont later be changed without the other).

Otherwise, if you wish to keep them separate, you can give one a different name.

Reply:
Yep still failing with this value `Specifies the OS type. Options are Linux and Windows. The default is Linux.`

I think it is better to not create a seperated one since it is a common definition for the stable service `aks` and `acs` and our new service in preview `osa`. 

It is a straighforward definition just with `Windows` and `Linux` as choice.

Should I change this description also in the stable definition ? I am just woried to impact the stable services.

Reply:
Aaah gotcha - yes if this is already defined then the description will need to be the same for each. It's not really a big deal - you can leave it as is for now to be consistent with the stable spec :)